src/escalation/jira_escalation.py

```diff
@@ -0,0 +1,226 @@
+        Args:
+          jira_query (string): JQL query string to get the issues
+          slack_channel (str): slack channel name to send notifications
+        """
+
+        issues_list = self.jira.search_issues(jql_query=jira_query)
+        LOGGER.info(f" Jira issues: {issues_list}")
+        # LOGGER.info(f"-------------------------Jira Issues----------------------------")
+        test_email = "vshakya@redhat.com"
+        for issue in issues_list:
+            jira_issue = self.jira.get_issue_by_id_or_key_with_changelog(issue)
+
+
+
+            LOGGER.info(f" ---------------jira issue: {jira_issue.key}--- Status : {jira_issue.fields.status.name}  ---------------")
+
+            if jira_issue.fields.assignee is None:
+                LOGGER.info(
+                    f" No assignee found for this jira issue : {jira_issue.key}"
+                )  # to do : send notification in slack to mpiit team for followup
+                issue_url = f'<https://issues.stage.redhat.com/browse/{jira_issue.key}|{jira_issue.key}>'
+                message = f"No assignee found for this jira issue :  {issue_url} \n please provide an update"
+                self.send_slack_notification(self.slack_client,slack_channel,message) #uncomment this for testing
+                # self.send_slack_notification(
+                #     self.slack_client, slack_channel, message
+                # )  # comment this for testing
+
+                continue
+
+            status_changed_date = None
+
+            if jira_issue.fields.status.name == "ACK":
+                change_log = jira_issue.changelog
+                status_changed_date = None
+                for history in change_log.histories:
+                    for item in history.items:
+                        # LOGGER.info(f" item: {item.field}")
+                        if item.field == "status":
+                            status_changed_date = datetime.datetime.strptime(history.created, "%Y-%m-%dT%H:%M:%S.%f%z")
+                            # LOGGER.info(f"status_changed_date: {status_changed_date}")
+                        break
+                    if status_changed_date:
+                        break
+
+            comments = jira_issue.fields.comment.comments
+
+            updated_date = datetime.datetime.strptime(jira_issue.fields.updated, "%Y-%m-%dT%H:%M:%S.%f%z")
+            assignee_comment_created_date = None
+            last_updated_time = None
+
+            assignee = jira_issue.fields.assignee
+            # assignee_email = assignee.emailAddress
+            # assignee_email = "vshakya@redhat.com"
+
+            # check if issue is updated within escalation threshold
+
+            for comment in comments:
+                if comment.author.emailAddress == assignee.emailAddress:
+                    assignee_comment_created_date = datetime.datetime.strptime(
+                        comment.created, "%Y-%m-%dT%H:%M:%S.%f%z"
+                    )
```

Comment:
Another possible checked be verifying if comments come from other than PQE: from other PQE team member/manager while PQE is unavailable

Reply:
can considered for future improvements but at this point we don't have PQE team info to implement this validation